File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.cpp
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        for(int i=0;i<nums.size();i++)
        {
            int max=nums[0];
            for(int j=0;j<=i;j++)
            {
                if(nums[j]>max)
                {
                    max=nums[j];
                }
            }
            int min=nums[i];
            for(int k=i;k<nums.size();k++)
            {
                if(nums[k]<min)
                {
                    min=nums[k];
                }
            }
            if((max-min)<=k)
            {
                return i;
            }
        }
        return -1;
    }
};

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
        for i in range(len(nums)):
            max=nums[0]
            for j in range(0,i+1):
                if(nums[j]>max):
                    max=nums[j]
            min=nums[i]
            for l in range(i,len(nums)):
                if(nums[l]<min):
                    min=nums[l]
            if((max-min)<=k):
                return i
        return -1
